klippy/extras/homing_heaters.py

```diff
@@ -0,0 +1,54 @@
+# Heater handling on homing moves
+#
+# Copyright (C) 2016-2018  Kevin O'Connor <kevin@koconnor.net>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+class HomingHeaters:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.printer.register_event_handler("homing:move_begin",
+                                            self.handle_homing_move_begin)
+        self.printer.register_event_handler("homing:move_end",
+                                            self.handle_homing_move_end)
+        self.heaters_to_disable = config.get("heaters", "")
+        self.steppers_needing_quiet = config.get("steppers", "")
+        self.last_target = {}
+        self.pheater = self.printer.lookup_object('heater')
+
+    def check_eligible(self, endstops):
+        flaky_steppers = [n.strip()
+                         for n in self.steppers_needing_quiet.split(',')]
+        if flaky_steppers == [""]:
+            return True
+        endstop_instances, dummy = zip(*endstops)
+        steppers_being_homed = list()
+        for n in endstop_instances:
+            steppers_being_homed = steppers_being_homed + n.get_steppers()
+        steppernames_being_homed = list()
+        for n in steppers_being_homed:
+            steppernames_being_homed.append(n.get_name())
+        return any(x in flaky_steppers for x in steppernames_being_homed)
+    def bld_heater_list(self):
+        heaters = [n.strip() for n in self.heaters_to_disable.split(',')]
+        if heaters == [""]:
+            heaters = self.pheater.available_heaters
```

Comment:
The convention in Klipper is to not reach into any member variable of an external class (that is any class not in the same .py file).  This mentioned briefly in https://www.klipper3d.org/Code_Overview.html#adding-a-host-module .  So, to get at the available heaters, either add a `get_all_heaters()` method to the PrinterHeaters class, or use `self.pheater.get_status()['available_heaters']`.

Reply:
Didn't know if I should minimize the places where changes are to be made. Adding a `get_all_heaters()` method, I would prefer that if it is allowed.

Reply:
commit 64409895fa4a132497060968d5bf24522a003e10